File: src/features/canvas/models/image/qianhai/gemini31FlashImagePreview.ts
import { createQianhaiImagePreviewModel } from './shared';

export const QIANHAI_GEMINI_3_1_FLASH_IMAGE_PREVIEW_MODEL_ID =
  'qianhai/gemini-3.1-flash-image-preview';

export const imageModel = createQianhaiImagePreviewModel({
  id: QIANHAI_GEMINI_3_1_FLASH_IMAGE_PREVIEW_MODEL_ID,
  displayName: 'Gemini 3.1 Flash Image Preview',
  description: '千海 · Gemini 3.1 Flash Image Preview 图像生成与编辑',
  requestModel: 'gemini-3.1-flash-image-preview',
});

File: src-tauri/src/ai/providers/qianhai/models/gemini_3_1_flash_image_preview.rs
use super::shared::GoogleCompatibleImagePreviewAdapter;

inventory::submit! {
    crate::ai::providers::qianhai::models::RegisteredQianhaiModel {
        build: || Box::new(GoogleCompatibleImagePreviewAdapter::new(
            "qianhai/gemini-3.1-flash-image-preview",
            &[
                "qianhai/gemini-3.1-flash-image-preview",
                "gemini-3.1-flash-image-preview",
            ],
        )),
    }
}
